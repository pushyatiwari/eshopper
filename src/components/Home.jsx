/* eslint-disable no-console */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/no-unused-state */
/* eslint-disable no-use-before-define */
import React from 'react';
import './Home.css';
import PropTypes from 'prop-types';
import Product from './Product';

function RenderCards(onIncrement, onDecrement, description) {
  return description.map((product) => (
    <Product
      key={product.id}
      description={product}
      onIncrement={() => onIncrement(product.id)}
      onDecrement={() => onDecrement(product.id)}
    />
  ));
}

function Home({
  onIncrement, onDecrement, description,
}) {
  return (
    <div className="container">
      <div className="rendercards">
        {RenderCards(onIncrement, onDecrement, description)}
      </div>
    </div>
  );
}

Home.propTypes = {
  description: PropTypes.objectOf(PropTypes.any).isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};
RenderCards.propTypes = {
  description: PropTypes.objectOf(PropTypes.any).isRequired,
  cartCount: PropTypes.number.isRequired,

};

export default Home;
// https://5.imimg.com/data5/EX/QK/MY-37427162/selection_008-500x500.png
