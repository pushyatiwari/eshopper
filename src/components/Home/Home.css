.rendercards{
display: flex;
align-content: stretch;
flex-wrap: wrap;
}
.container{
    display: flex;
    flex-direction: column;
}
@media only screen and (max-width: 20rem) {
    .rendercards{
        flex-direction: column;
        align-content: stretch;
    }
}