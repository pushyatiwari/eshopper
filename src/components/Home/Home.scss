@import '../../styles//index.scss';

.rendercards{
display: flex;
align-content: stretch;
flex-wrap: wrap;
}
.container{
    display: flex;
    flex-direction: column;
}
@media only screen and (max-width: 20rem) {
    .rendercards{
        flex-direction: column;
        align-content: stretch;
    }
}
.darkTheme{
background-color: $theme-color-dark;
}
.lightTheme{
    background-color: $theme-color-light;
    }