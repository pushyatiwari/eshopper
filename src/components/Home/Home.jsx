import React, { useContext } from 'react';
import './Home.scss';
import PropTypes from 'prop-types';
import Product from '../Product/Product';
import { ThemeContext } from '../../themeContext';

const RenderCards = (onIncrement, onDecrement, description) => description.map((product) => (
  <Product
    key={product.id}
    description={product}
    onIncrement={() => onIncrement(product.id)}
    onDecrement={() => onDecrement(product.id)}
  />
));

const Home = ({ onIncrement, onDecrement, description }) => {
  const theme = useContext(ThemeContext);
  return (
    <div className={theme === 'dark' ? 'darkTheme container' : 'whiteTheme container'}>
      <div className="rendercards">
        {RenderCards(onIncrement, onDecrement, description)}
      </div>
    </div>
  );
};

const configShape = {
  id: PropTypes.string,
  name: PropTypes.string,
  price: PropTypes.number,
  count: PropTypes.number,
  src: PropTypes.string,
};
Home.propTypes = {
  description: PropTypes.arrayOf(PropTypes.shape(configShape)).isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};
RenderCards.propTypes = {
  description: PropTypes.arrayOf(PropTypes.shape(configShape)).isRequired,
  cartCount: PropTypes.number.isRequired,

};

export default Home;
// https://5.imimg.com/data5/EX/QK/MY-37427162/selection_008-500x500.png
