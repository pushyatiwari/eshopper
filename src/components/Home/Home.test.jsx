import {
  render, screen, waitFor,
} from '@testing-library/react';
import React from 'react';
import Home from './Home';

describe(Home.name, () => {
  const mockFunction = {
    onIncrement: jest.fn(),
    onDecrement: jest.fn(),
  };
  const mockData = {
    'Fruits & Vegatables': [{
      category: 'Fruits & Vegatables', count: 20, countInCart: 0, id: 1, name: 'apple', price: 120, src: 'https://images.unsplash.com/photo-1571771894821-ce9b6c11b08e?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1000&q=80',
    }],
    'Household Items': [{
      category: 'Household Items', count: 3, countInCart: 0, id: 2, name: 'table cloth', price: 200, src: 'https://images.unsplash.com/photo-1571771894821-ce9b6c11b08e?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1000&q=80',
    }],
  };

  test('should render Home', async () => {
    const renderHome = render(
      <Home
        description={mockData}
        onIncrement={mockFunction.onIncrement}
        onDecrement={mockFunction.onDecrement}
      />,
    );
    expect(renderHome).toMatchSnapshot();
  });
});
