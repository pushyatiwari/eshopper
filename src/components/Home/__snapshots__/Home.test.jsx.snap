// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render Home 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="whiteTheme container"
        data-testid="home-route"
      >
        <div
          class="rendercards"
        >
          <div
            class="productContainer"
            data-testid="product"
          >
            <div
              class="product"
            >
              <p>
                Fruits & Vegatables
              </p>
              <img
                alt="product"
                src="https://images.unsplash.com/photo-1571771894821-ce9b6c11b08e?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1000&q=80"
              />
              <p
                class="productDescription"
              >
                apple
              </p>
              <p
                class="productDescription"
              >
                1 kg
              </p>
            </div>
            <div
              class="description"
            >
              <p>
                MRP:
                120
              </p>
              <div
                class="basketItems"
              >
                <button
                  type="button"
                >
                  -
                </button>
                0
                <button
                  type="button"
                >
                  +
                </button>
              </div>
            </div>
          </div>
          <div
            class="productContainer"
            data-testid="product"
          >
            <div
              class="product"
            >
              <p>
                Household Items
              </p>
              <img
                alt="product"
                src="https://images.unsplash.com/photo-1571771894821-ce9b6c11b08e?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1000&q=80"
              />
              <p
                class="productDescription"
              >
                table cloth
              </p>
              <p
                class="productDescription"
              >
                1 kg
              </p>
            </div>
            <div
              class="description"
            >
              <p>
                MRP:
                200
              </p>
              <div
                class="basketItems"
              >
                <button
                  type="button"
                >
                  -
                </button>
                0
                <button
                  type="button"
                >
                  +
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="whiteTheme container"
      data-testid="home-route"
    >
      <div
        class="rendercards"
      >
        <div
          class="productContainer"
          data-testid="product"
        >
          <div
            class="product"
          >
            <p>
              Fruits & Vegatables
            </p>
            <img
              alt="product"
              src="https://images.unsplash.com/photo-1571771894821-ce9b6c11b08e?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1000&q=80"
            />
            <p
              class="productDescription"
            >
              apple
            </p>
            <p
              class="productDescription"
            >
              1 kg
            </p>
          </div>
          <div
            class="description"
          >
            <p>
              MRP:
              120
            </p>
            <div
              class="basketItems"
            >
              <button
                type="button"
              >
                -
              </button>
              0
              <button
                type="button"
              >
                +
              </button>
            </div>
          </div>
        </div>
        <div
          class="productContainer"
          data-testid="product"
        >
          <div
            class="product"
          >
            <p>
              Household Items
            </p>
            <img
              alt="product"
              src="https://images.unsplash.com/photo-1571771894821-ce9b6c11b08e?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1000&q=80"
            />
            <p
              class="productDescription"
            >
              table cloth
            </p>
            <p
              class="productDescription"
            >
              1 kg
            </p>
          </div>
          <div
            class="description"
          >
            <p>
              MRP:
              200
            </p>
            <div
              class="basketItems"
            >
              <button
                type="button"
              >
                -
              </button>
              0
              <button
                type="button"
              >
                +
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
